### PR TITLE
You can now hide papers and cards in vents

### PR DESCRIPTION
**Pull request by Robertha89**
:cl: PPI
rscadd: Adds the ability to hide papers in vents. You can now leave a romantic love letter, exchange information in secret, or hide papers infused with the power of nar'sie from sight.
/:cl:

### CODE REVIEW

File: code/ATMOSPHERICS/components/unary_devices/vent_pump.dm [hunk 1]
```diff
@@ -339,6 +340,39 @@
...
+				if (do_after(user, 20, target = src))
+					open = 0
+					user.visible_message("[user] screwdrivers the vent shut.", "You screwdriver the vent shut.", "You hear a screwdriver.")
+			else
+				user << "\blue Now opening the vent."
+				if (do_after(user, 20, target = src))
+					open = 1
+					user.visible_message("[user] screwdrivers the vent shut.", "You screwdriver the vent shut.", "You hear a screwdriver.")
+		return
+	if(istype(W, /obj/item/weapon/paper/))
```
Comment (Crazylemon64) [inline code/ATMOSPHERICS/components/unary_devices/vent_pump.dm]: I don't think this will accept normal paper, you'll want to remove that final slash.

**Comment by Earthdivine** [comment]
Now if only it was easier to get to Brig, would be cool to leave notes for incoming prisoners that theres a stash.

**Comment by NTSAM** [comment]
*Writes love letter, places in QM's vent, gets arrested for trespassing*
:+1:

**Comment by fludd12** [comment]
Jeez, you're on a roll, aren't you? It's the little stuff like this and head patting that adds a lot in the long run, I think.

File: code/ATMOSPHERICS/components/unary_devices/vent_pump.dm [hunk 2]
```diff
@@ -339,6 +340,39 @@
 		else
 			user << "<span class='notice'>You need more welding fuel to complete this task.</span>"
 			return 1
+	if(istype(W, /obj/item/weapon/screwdriver))
+		if(!welded)
+			if(open)
+				user << "\blue Now closing the vent."
```
Comment (TheDZD) [inline code/ATMOSPHERICS/components/unary_devices/vent_pump.dm]: Span classes, not color macros.

File: code/ATMOSPHERICS/components/unary_devices/vent_pump.dm [hunk 3]
```diff
@@ -339,6 +340,39 @@
...
+				if (do_after(user, 20, target = src))
+					open = 0
+					user.visible_message("[user] screwdrivers the vent shut.", "You screwdriver the vent shut.", "You hear a screwdriver.")
+			else
+				user << "\blue Now opening the vent."
```
Comment (TheDZD) [inline code/ATMOSPHERICS/components/unary_devices/vent_pump.dm]: Same here.

**Comment by Robertha89** [comment]
this is intentionally meant as a buff too most antags, and cult.
But yes, it could probably be used to send love letters and the such. Which i did not even consider.

File: code/ATMOSPHERICS/components/unary_devices/vent_pump.dm (hunk 1 above)
Comment (Robertha89) [inline code/ATMOSPHERICS/components/unary_devices/vent_pump.dm]: it will, just tested.

Comment (Crazylemon64) [inline code/ATMOSPHERICS/components/unary_devices/vent_pump.dm]: Alright! Looks like I'll need to remember that the slash doesn't do anything, in the future - for some reason, I thought it did.

Comment (Fox-McCloud) [inline code/ATMOSPHERICS/components/unary_devices/vent_pump.dm]: It's still non-standard and out of line with everything else.

**Comment by KasparoVy** [comment]
>leaves super sekrit note in vent
>pdas Chuck Lefuck "i feel a draft :^)"
>checks the vent
>"git gud"
>exploding paper, gadget style

**Comment by Earthdivine** [comment]
@KasparoVy Put C4 on paper, put it in vent, run off. :v

**Comment by KasparoVy** [comment]
sortof what i was eluding to

**Comment by fludd12** [comment]
[This is relevant](http://www.giantitp.com/comics/oots0221.html).

**Comment by theColdflame** [comment]
does cult really need a buff in its current state?

File: code/ATMOSPHERICS/components/unary_devices/vent_pump.dm [hunk 4]
```diff
@@ -339,6 +340,39 @@
...
+				user << "<span class='notice'> Now opening the vent.</span>"
+				if (do_after(user, 20, target = src))
+					open = 1
+					user.visible_message("[user] screwdrivers the vent shut.", "You screwdriver the vent shut.", "You hear a screwdriver.")
+		return
+	if(istype(W, /obj/item/weapon/paper))
+		if(!welded)
+			if(open)
+				user.drop_item(W)
+				W.loc = src
```
Comment (Fox-McCloud) [inline code/ATMOSPHERICS/components/unary_devices/vent_pump.dm]: Don't set loc directly; use forcemove.

Comment (TheDZD) [inline code/ATMOSPHERICS/components/unary_devices/vent_pump.dm]: `W.forceMove(src)`

File: code/ATMOSPHERICS/components/unary_devices/vent_pump.dm [hunk 5]
```diff
@@ -339,6 +340,39 @@
...
+			if(!open)
+				user << "You can't shove that down there when it is closed"
+		else
+			user << "The vent is welded."
+		return
+	if(istype(W, /obj/item/weapon/card))
+		if(!welded)
+			if(open)
+				user.drop_item(W)
+				W.loc = src
```
Comment (TheDZD) [inline code/ATMOSPHERICS/components/unary_devices/vent_pump.dm]: `W.forceMove(src)`

**Comment by TheDZD** [comment]
:+1: to the idea of hiding notes inside of air vents. Not so certain on cards though, especially since you can even hide emags that way.

File: code/ATMOSPHERICS/components/unary_devices/vent_pump.dm [hunk 6]
```diff
@@ -349,6 +383,12 @@
 
 	return ..()
 
+/obj/machinery/atmospherics/unary/vent_pump/attack_hand()
+	if(!welded)
+		if(open)
+			for(var/obj/item/weapon/W in src)
+				W.loc = src.loc
```
Comment (Robertha89) [inline code/ATMOSPHERICS/components/unary_devices/vent_pump.dm]: What about here?

Comment (TheDZD) [inline code/ATMOSPHERICS/components/unary_devices/vent_pump.dm]: `W.forceMove(get_turf(src))`

**Comment by Robertha89** [comment]
@TheDZD I understand the reluctance, but it will make it easier for tators to hide emags, and they don't need to spend loads of tc on getting the ability to secretly carry it everywhere.
Anyone can find if they check, thus it's own weakness.

**Comment by TheDZD** [comment]
@Robertha89 True, although in that sense it's very similar to using a smuggler's pack, which you can stuff things into and hide under a floor tile.

**Comment by Robertha89** [comment]
each has it's own pro and cons.
Implant
+hidden
+max weight 6
-2 slots
-high cost (8)
-can be removed
-only normal sized or smaller
Smuggler
+hide in floor
+hide in pocket(not sure if this is still a thing)
+extra carry
+max weight 15
+can hide in backpack
+cheap (2)
+comes with a crowbar
Hide in vent
+they are everywhere
+unlikely to be found
+stealthy to hide
-easier to be found by accident
-not stealthy to grab
-takes time
-can be surrounded by people

Might have missed a few, but each got it's advantages and whatnot, this just gives it to everyone, irregardless of antag or not.

**Comment by Iamgoofball** [comment]
HOLY FUCKING SNOWFLAKE BATMAN WHAT THE /FUCK/

because atmos code needed even more snowflake

:-1:

**Comment by fludd12** [comment]
With all due respect, @Iamgoofball , your past few comments on Paradise pulls have been nothing but criticism with very little in regards to how to fix these issues. I'm sure @Robertha89 would appreciate some constructive criticism, but just saying "snowflake code" doesn't really help anyone.

**Comment by Fox-McCloud** [comment]
:-1: to being able to hide cards.

**Comment by Iamgoofball** [comment]
@fludd12 Let's see:
1. This serves no practical purpose at all. It's the worst possible way of delivering or hiding a paper ever. Paper also is the least important thing on the server.
2. It uses FORCEMOVE. TO PUT THE PAPER IN. WHY.
3. This adds 40 lines to atmospherics, which is already messy enough, cluttering it up with a really useless feature that shouldn't exist in the first place. This didn't need 40 lines.
4. It's hardcoded for paper on the insertion end but supports anything on the removal end. Why? Just allow inserting any small pocket sized object, then this feature will actually be useful.

**Comment by Spacemanspark** [comment]
Because having no practical purpose means it shouldn't be here, right?
If that's the case, then we should remove toys/ arcade machines/ hugging/ syndiballoons/ ten billion other things.
The 'snowflakecode' holds some weight, though, if it's an issue.
I do like the concept behind this PR.

**Comment by Crazylemon64** [comment]
> 2. It uses FORCEMOVE. TO PUT THE PAPER IN. WHY.
Err, the problem here? Don't we use forcemove for every other purpose of moving things around?

**Comment by Fox-McCloud** [comment]
@Robertha89 

Going to remove the card hiding?

**Comment by Fox-McCloud** [comment]
@Robertha89 

Status?

**Comment by Robertha89** [comment]
well, the card hiding is a huge part of it, but eh, if paper in vents alone comes along dashingly, it can be expanded in the future.

**Comment by Fox-McCloud** [comment]
@TheDZD 

Waiting for your final decision.

**Pull request merged**